### ISSUE TITLE
[STORY: E001-S001] Create Markdown Issue and Pull Request Templates for Agile Project Management in Software Development

### ISSUE THREAD

**Issue by antonio-crispino** [antonio-crispino/.github#3]
# Story: [E001](https://github.com/antonio-crispino/.github/issues/1 "Parent Issue (Epic)")-S001 <span id="s-story">

**As a** project collaborator, **I want** to be able to create issues and pull requests using structured templates, **so that** the information included in specific issue types and pull requests is standardized.

## Table of Contents <span id="s-table-of-contents">

1. [Description :memo:](#s-description "A detailed description of the Story. This includes any relevant context, requirements, and constraints.")
2. [Priority :exclamation:](#s-priority "The priority of the Story (MoSCoW method).")
3. [User Story Points (USP) :star:](#s-user-story-points-usp "The estimated effort required to complete the Story.")
4. [Dependencies :chains:](#s-dependencies "A list of dependencies required for the completion of the Story.")
5. [Acceptance Criteria :heavy_check_mark:](#s-acceptance-criteria "A list of conditions that must be met for the Story to be considered complete. This can use the Given-When-Then (GWT) format and other statement formats if appropriate.")
6. [Subtasks :clipboard:](#s-subtasks "A list of subtasks required to complete the Story. These do not need to have their own Subtask issues, but they could.")
7. [Parent Issue (Epic) :link:](#s-parent-issue-epic "The parent issue (Epic) for this Story, if it has one.")
8. [Resources :books:](#s-resources "A list of useful resources/links for the Story.")
9. [Notes :pencil2:](#s-notes "A list of additional information about the Story.")

## 1 - Description :memo: <span id="s-description">

> _A detailed description of the Story. This includes any relevant context, requirements, and constraints._

This story involves creating markdown issue and pull request templates for agile project management in software development. One general issue template for each of the following issues is to be created: Epics, Stories, Tasks, Bugs, and Subtasks. Also, one general pull request (PR) template is to be created. Research and best judgement should be used when creating the templates. The Jira templates should be used as a starting point (links provided in resources section below).

## 2 - Priority :exclamation: <span id="s-priority">

> _The priority of the Story (MoSCoW method). Select one of the following priorities:_
>
> 1. :sunny: `must-have`
> 2. :large_orange_diamond: `should-have`
> 3. :large_blue_diamond: `could-have`
> 4. :no_entry: `won't-have`

`must-have`

## 3 - User Story Points (USP) :star: <span id="s-user-story-points-usp">

> _The estimated effort required to complete the Story. This is estimated using user story points (USP). The number of user story points allocated is selected from the following Fibonacci-like sequence: [1, 2, 3, 5, 8, 13, 20, 40, 100]._

13 USP

## 4 - Dependencies :chains: <span id="s-dependencies">

> _A list of dependencies required for the completion of the Story._

1. **The Story** cannot start until Task [E001-T001](https://github.com/antonio-crispino/.github/issues/2) is completed.

## 5 - Acceptance Criteria :heavy_check_mark: <span id="s-acceptance-criteria">

> _A list of conditions that must be met for the Story to be considered complete. This can use the Given-When-Then (GWT) format and other statement formats if appropriate._

1. **Given** the GitHub website is open, **when** a contributor creates a new issue, **then** there are Epic, Story, Task, Bug, and Subtask issue template options available.
2. **Given** an issue is complete, **when** a PR is made, **then** a general PR template is present.

## 6 - Subtasks :clipboard: <span id="s-subtasks">

> _A list of subtasks required to complete the Story. These do not need to have their own Subtask issues, but they could._

- [x] 1. Create Epic issue template.
- [x] 2. Create Story issue template.
- [x] 3. Create Task issue template.
- [x] 4. Create Bug issue template.
- [x] 5. Create Subtask issue template.
- [x] 6. Create config file for the issue templates. This should enable collaborators to create custom issues along side the provided templates, which is the default configuration.
- [x] 7. Create Pull Request (PR) template.

## 7 - Parent Issue (Epic) :link: <span id="s-parent-issue-epic">

> _The parent issue (Epic) for this Story, if it has one._

[Epic E001](https://github.com/antonio-crispino/.github/issues/1 "Parent Issue (Epic)"): Create Default Community Health Files

## 8 - Resources :books: <span id="s-resources">

> _A list of useful resources/links for the Story._

1. [Jira Templates](https://www.jira-templates.com/issues)
2. [Jira Epic Template](https://www.jira-templates.com/issues/epic-template)
3. [Jira Story Template](https://www.jira-templates.com/issues/story-template)
4. [Jira Task Template](https://www.jira-templates.com/issues/task-template)
5. [Jira Bug Template](https://www.jira-templates.com/issues/bug-template)
6. [About GitHub Issue and Pull Request Templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates)
7. [Creating GitHub Issue Templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/manually-creating-a-single-issue-template-for-your-repository)
8. [Configuring the GitHub Issue Template Chooser](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#configuring-the-template-chooser)

## 9 - Notes :pencil2: <span id="s-notes">

> _A list of additional information about the Story._

None
